src/credentials/attestation_verifier/DeviceAttestationVerifier.h

```diff
@@ -222,6 +220,25 @@ class DeviceAttestationVerifier
         uint16_t productId;
     };
 
+    // Copies the bytes passed to it, and holds the PAI, DAC, and CD for additional verification step
+    class AttestationDeviceInfo
+    {
+    public:
+        AttestationDeviceInfo(const AttestationInfo & attestationInfo);
+        AttestationDeviceInfo(const ByteSpan & attestationElementsBuffer, const ByteSpan paiDerBuffer, const ByteSpan dacDerBuffer);
+
+        ~AttestationDeviceInfo();
+
+        ByteSpan mPaiDerBuffer;       // Buffer containing the PAI certificate from device in DER format.
+        ByteSpan mDacDerBuffer;       // Buffer containing the DAC certificate from device in DER format.
+        Optional<ByteSpan> mCdBuffer; // Buffer containing the certificate declaration from device.
```

Comment:
Seems like you don't want to provide direct access since these can be modified. Make these private and have inline getters?

Reply:
@anush-apple How do you think that should look like?

Reply:
Make them private and then have:

ByteSpan paiDerBuffer() const  { return mPaiDerBuffer; }